### PR TITLE
Add GitHub Action to Auto Update Change logs

### PR DESCRIPTION
**Pull request by SudharakaP**
This adds a GitHub action to auto update the change logs.

Fixes https://github.com/opencollective/opencollective/issues/2984

### CODE REVIEW

**Comment by SudharakaP** [comment]
@Betree : Yes, I've [tested this out locally in my fork](https://github.com/SudharakaP/opencollective-changelog/runs/592645504?check_suite_focus=true) and it seems to work fine. I think the checkout@v2 action has a token to commit by default. All we got to have it the line, 

`run: git config --global user.email "actions@github.com" && git config --global user.name "GitHub Actions"`

to configure the `user.email` and `user.name` on git. And these values I've added at random. Please feel free to suggest something else for them if you prefer. :smile:

**Comment by Betree** [comment]
The [CI example you posted](https://github.com/SudharakaP/opencollective-changelog/runs/592645504) looks great, but I can't seem to find the pushed commit associated with it. Do you have an URL pointing to it?

**Comment by SudharakaP** [comment]
@Betree : Notice the line; `d17cda0..b2b27dd  update-changelog-automatically -> update-changelog-automatically` which is where it was pushed in the logs. To test this I had to change the line; 

https://github.com/opencollective/opencollective-changelog/blob/0edc048de3767df8b85126237b5220d6225386ca/scripts/commit_logs.sh#L5

to point to the branch `update-changelog-automatically` instead of master. And I removed that commit later since I don't want it to be checked in (the updated changelogs). Would you like to me create a separate branch and do a demonstration? :smile:

**Comment by SudharakaP** [comment]
@Betree : Okay, so I've created a demo; 

https://github.com/SudharakaP/opencollective-changelog/runs/595832534?check_suite_focus=true

and the committed push is;

https://github.com/SudharakaP/opencollective-changelog/commit/54e687d02810167307aede816dd64cd6facfe913

Note that in this branch I had to change the [Git command accordingly](https://github.com/opencollective/opencollective-changelog/commit/b731e86d3e6f727c848e54150bb605f41038c06b); which is the reason why I didn't want to check that in.

**Comment by SudharakaP** [comment]
@Betree : You are welcome. I just made a commit to update the readme. Take a look and let me know if that's fine with you. 

Also I noticed that the cron job schedule had a small mistake; it used to be `cron: '0 * * * *'` which I copied from https://github.com/opencollective/opencollective/issues/2984; which will run it every hour. I changed it to `cron: '0 0 * * *'` since I think what you really want is run it every day at midnight. Please confirm if this is the case. :smile:

**Comment by Betree** [comment]
Every hour was fine, but we can start with every day. I'm fine with both :slightly_smiling_face:

File: .github/workflows/update-changelog.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: git config --global user.email "actions@github.com" && git config --global user.name "GitHub Actions"
```
Comment (znarf) [inline .github/workflows/update-changelog.yml:23]: Could you change to 

```suggestion
      - run: git config --global user.email "ci@opencollective.com" && git config --global user.name "Open Collective CI"
```

Comment (SudharakaP) [reply .github/workflows/update-changelog.yml]: @znarf : Done :smile:

**Comment by Betree** [comment]
Let's try this :rocket:

**Comment by SudharakaP** [comment]
@Betree : Wonderful. Feel free to ping me if you see any issues. :smile:

**Comment by Betree** [comment]
It worked, commit: https://github.com/opencollective/opencollective-changelog/commit/b67056f392634fd0d6e7ea28bf56174c57706013

The auto-deploy worked too: https://changelog.opencollective.com/

**Comment by SudharakaP** [comment]
@Betree : Yay; that's wonderful. :+1:

**Pull request merged**